src/analysis/inner_product_space/cross_product.lean

```diff
@@ -0,0 +1,181 @@
+
+The cross-product is constructed here using the volume form, a 3-form on `E` determined uniquely by
+the orientation and inner product structure. The partial evaluation of the volume form o on `u` and
+`v` gives an element o(u, v, ⬝) of the dual of `E`.  This can then be identified with `E` itself
+using the canonical isometry (induced by the inner product) between `E` and its dual.
+
+See `matrix.cross_product` for a construction of the cross-product in co-ordinates. -/
+@[irreducible] def cross_product : E →ₗ[ℝ] E →ₗ[ℝ] E :=
+begin
+  let to_dual : E ≃ₗ[ℝ] (E →ₗ[ℝ] ℝ) :=
+    (inner_product_space.to_dual ℝ E).to_linear_equiv ≪≫ₗ linear_map.to_continuous_linear_map.symm,
+  let z : alternating_map ℝ E ℝ (fin 0) ≃ₗ[ℝ] ℝ :=
+    alternating_map.const_linear_equiv_of_is_empty.symm,
+  let y : alternating_map ℝ E ℝ (fin 1) →ₗ[ℝ] E →ₗ[ℝ] ℝ :=
+    (linear_map.llcomp ℝ E (alternating_map ℝ E ℝ (fin 0)) ℝ z) ∘ₗ
+      alternating_map.curry_left_linear_map,
+  let y' : alternating_map ℝ E ℝ (fin 1) →ₗ[ℝ] E :=
+    (linear_map.llcomp ℝ (alternating_map ℝ E ℝ (fin 1)) (E →ₗ[ℝ] ℝ) E to_dual.symm) y,
+  let u : alternating_map ℝ E ℝ (fin 2) →ₗ[ℝ] E →ₗ[ℝ] E :=
+    (linear_map.llcomp ℝ E (alternating_map ℝ E ℝ (fin 1)) _ y') ∘ₗ
+      alternating_map.curry_left_linear_map,
+  exact u ∘ₗ (alternating_map.curry_left_linear_map o.volume_form),
+end
+
+local infixl ` ×₃ `: 74 := o.cross_product
+
+/-- Defining property of the cross-product: for vectors `u`, `v`, `w`, the inner product of u × v
+with w is equal to o(u, v, w), where -/
+lemma inner_cross_product_apply (u v w : E) : ⟪u ×₃ v, w⟫ = o.volume_form ![u, v, w] :=
+by simp only [cross_product, linear_equiv.trans_symm, linear_equiv.symm_symm,
+  linear_isometry_equiv.to_linear_equiv_symm, alternating_map.curry_left_linear_map_apply,
+  linear_map.coe_comp, function.comp_app, linear_map.llcomp_apply,
+  linear_equiv.coe_coe, linear_equiv.trans_apply, linear_isometry_equiv.coe_to_linear_equiv,
+  linear_isometry_equiv.norm_map, submodule.coe_norm,
+  inner_product_space.to_dual_symm_apply, alternating_map.curry_left_apply_apply,
+  alternating_map.const_linear_equiv_of_is_empty_symm_apply,
+  eq_self_iff_true, linear_map.coe_to_continuous_linear_map', matrix.zero_empty]
+
+/-- The cross-product is antisymmetric: `x ×₃ y = - (y ×₃ x)`. -/
+lemma cross_product_swap (x y : E) : x ×₃ y = - (y ×₃ x) :=
+begin
+  apply ext_inner_right ℝ,
+  intros z,
+  rw [inner_neg_left, o.inner_cross_product_apply, o.inner_cross_product_apply],
+  convert o.volume_form.map_swap ![y, x, z] (_ : (0 : fin 3) ≠ 1),
+  { ext i,
+    fin_cases i; refl },
+  { norm_num }
+end
+
+/-- The cross-product is antisymmetric: the cross-product of a vector with itself is 0. -/
+lemma cross_product_apply_self (v : E) : v ×₃ v = 0 :=
+begin
+  apply ext_inner_right ℝ,
+  intros w,
+  rw [inner_cross_product_apply, inner_zero_left],
+  refine o.volume_form.map_eq_zero_of_eq ![v, v, w] _ (_ : (0 : fin 3) ≠ 1),
+  { simp },
+  { norm_num }
+end
```

Comment:
```suggestion
begin
  rw [←two_nsmul_eq_zero ℝ, two_nsmul],
  nth_rewrite 1 cross_product_swap,
  exact add_neg_self _,
end
```
Oliver's suggestion. `exact` could also be a `rw` if you prefer that, I did not really try to golf it any further.